### PR TITLE
MIN aggregation

### LINKED ISSUE #3: Add MIN aggregation

### PR DESCRIPTION
**Pull request by mcarmonaa**
Closes #3

### CODE REVIEW

File: sql/expression/aggregation.go [hunk 1]
```diff
@@ -107,3 +107,57 @@ func (e *First) Merge(buffer, partial sql.Row) {
...
 	return buffer[0]
 }
+
+// Min aggregation returns the smallest value of the selected column.
+// It implements the AggregationExpression interface
+type Min struct {
+	UnaryExpression
+}
+
+func NewMin(e sql.Expression) *Min {
```
Comment (erizocosmico) [inline sql/expression/aggregation.go]: the rest of the exported members are missing comments

Comment (mcarmonaa) [reply sql/expression/aggregation.go]: Done

File: sql/expression/aggregation_test.go [hunk 2]
```diff
@@ -96,3 +97,53 @@ func TestFirst_Eval(t *testing.T) {
 	c.Merge(b, b2)
 	assert.Equal(int32(1), c.Eval(b))
 }
+
+func TestMin_Name(t *testing.T) {
```
Comment (ajnavarro) [inline sql/expression/aggregation_test.go]: The same as MAX; can you add some tests with nulls please?

Comment (ajnavarro) [inline sql/expression/aggregation_test.go]: The same as MAX; can you add some tests with nulls please?

Corner cases here: http://www-cs-students.stanford.edu/~wlam/compsci/sqlnulls

Comment (mcarmonaa) [reply sql/expression/aggregation_test.go]: @ajnavarro since the NULL thing is an enhancement to be implemented #33, can I merge this PR?

**Comment by mcarmonaa** [comment]
@ajnavarro @erizocosmico as in #16, now `NULL` values are handled in the `MIN` node although when the #33 will be implemented, this logic could be useless.

**Comment by erizocosmico** [comment]
Rebase needed
